File: gpt-4-turbo/stage2/loop_construct_seq_clause.cpp
#include <iostream>
#include <stdlib.h>
#include <openacc.h>

#ifndef ACC_TESTSUITE_H
#define ACC_TESTSUITE_H
#define SEED 12345
#define NUM_TEST_CALLS 10

// Utility to check OpenACC runtime version for completeness; not essential for this test
void check_openacc_version(){
    int major, minor;
    acc_get_property(0, acc_device_not_host, acc_property_major_version, &major);
    acc_get_property(0, acc_device_not_host, acc_property_minor_version, &minor);
    std::cout << "Running with OpenACC Version: " << major << "." << minor << std::endl;
}

#endif // ACC_TESTSUITE_H

#ifndef T1
// T1:loop construct seq clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);
    const int size = 1024;
    int data[size];
    
    // Initialize data to 0
    for (int i = 0; i < size; ++i){
        data[i] = 0;
    }
    
    // Expected outcome: data[i] = i
    #pragma acc parallel num_gangs(1) vector_length(1)
    {
        #pragma acc loop seq
        for (int i = 0; i < size; ++i){
            data[i] = i;
        }
    }
    
    // Validate sequential execution
    for (int i = 0; i < size; ++i){
        if(data[i] != i){
            err++;
        }
    }

    return err;
}
#endif

int main(){
    check_openacc_version();  // Optional: Confirm OpenACC API version
    
    int failcode = 0;
    int failed = 0;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-4-turbo/stage2/loop_construct_seq_clause.f90
! Example code to test the OpenACC loop construct seq clause
! Make sure to compile with OpenACC support enabled in your Fortran compiler

! Mock include for standardizing the testsuite environment (replace with actual path or content)
INCLUDE "acc_testsuite.Fh"

! Function to test the OpenACC loop construct seq clause
! It returns .FALSE. if the test passes (no errors), .TRUE. otherwise
LOGICAL FUNCTION test1()
  USE OPENACC
  IMPLICIT NONE
  INTEGER :: i
  REAL :: A(100), B(100)
  REAL :: sum
  INTEGER :: errors
  
  errors = 0
  sum = 0.0
  
  ! Initialize arrays
  A = 1.0
  B = 2.0
  
  ! Apply the seq clause to ensure sequential execution
  !$acc parallel loop seq
  DO i = 1, 100
    A(i) = A(i) + B(i)
  END DO
  !$acc end parallel loop
  
  ! Verify the computation was done sequentially
  DO i = 1, 100
    IF (A(i) /= 3.0) THEN
      errors = errors + 1
    END IF
  END DO
  
  IF (errors .eq. 0) THEN
    test1 = .FALSE.
  ELSE
    test1 = .TRUE.
  END IF
END FUNCTION test1

PROGRAM main
  IMPLICIT NONE
  INTEGER :: failcode, testrun
  LOGICAL :: failed
  INCLUDE "acc_testsuite.Fh"
  
  LOGICAL :: test1
  
  failed = .FALSE.
  failcode = 0
  
  DO testrun = 1, NUM_TEST_CALLS
    failed = failed .or. test1()
  END DO
  IF (failed) THEN
    failcode = failcode + 2 ** 0
    failed = .FALSE.
  END IF
  
  CALL EXIT (failcode)
END PROGRAM main